sdk/js/src/Client.ts

```diff
@@ -121,9 +124,14 @@ export class DevCycleClient<
         this.registerVisibilityChangeHandler()
 
         this.onInitialized = new Promise((resolve, reject) => {
-            this.resolveOnInitialized = (value) => {
-                this._isInitialized = true
-                resolve(value)
+            this.resolveOnInitialized = (value, error) => {
```

Comment:
this function should probably be called `settleOnInitialized` to better reflect what it does now